### PR TITLE
GAT-24: Removed the force password change requirement

### PR DESCRIPTION
**Pull request by barbun**
- Swapped the default driver to Drupal API, additional profile is for Drush driver.
- Adjusted the build.xml to run two profiles (default and drush)
- Added a condition to maximizeWindow function to only act when using Selenium
- Added a custom function that logs the created users from all the scenarios with @beans tag
- Added a cleanup Password state function that hooks on afterUserCreate to cancel the forced password change rule from password_policy.
- Tweaked assertAccountCreated function to ensure that the user roles are getting applied properly.
- Adjusted the location for the failure screenshot.
- Added a cleanup function for beans.
- Removed the @drupal tag the scenarios as Drupal API is now used by default.
- Removed all steps definitions that had "..and don't force password change" etc.

### CODE REVIEW

File: build/tests/behat/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -27,31 +35,52 @@ public function __construct() {
   /**
    * Set default browser window size to maximum.
    *
-   * @BeforeScenario @drupal
+   * @BeforeScenario
    */
   public function maximizeWindow() {
+    $driver = $this->getSession()->getDriver();
+    if (!($driver instanceof Selenium2Driver)) {
+      return;
```
Comment (ivrh) [inline build/tests/behat/bootstrap/FeatureContext.php]: Would it make more sense to throw an exception? You are silently returning at the moment

File: build/tests/behat/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -27,31 +35,52 @@ public function __construct() {
...
+   * Log the user IDs of the users that create beans.
+   *
+   * Collect user IDs later to be used for cleaning up the test beans.
+   *
+   * @afterUserCreate
+   */
+  public function logBeanUsers(EntityScope $scope) {
+    // Retrieve the user.
+    $user = $scope->getEntity();
+    if (isset($user->uid)) {
```
Comment (ivrh) [inline build/tests/behat/bootstrap/FeatureContext.php]: Prefer to use 'empty()` across the board. If $user is empty, $user->uid will trigger PHP notice.

`if (!empty($user->uid)) {`
will work without a notice.

File: build/tests/behat/bootstrap/FeatureContext.php [hunk 3]
```diff
@@ -27,31 +35,52 @@ public function __construct() {
...
-      'password_policy_history',
-    );
-
-    foreach ($tables as $table) {
-      db_delete($table)
-        ->condition('uid', $uid)
+  public function cleanUserPasswordState(EntityScope $scope) {
+    // Retrieve the user.
+    $user = $scope->getEntity();
+    if (isset($user->uid)) {
```
Comment (ivrh) [inline build/tests/behat/bootstrap/FeatureContext.php]: Same, recommend using `if (!empty())`

File: build/tests/behat/bootstrap/FeatureContext.php (hunk 2 above)
Comment (barbun) [inline build/tests/behat/bootstrap/FeatureContext.php]: Agreed, thanks!

File: build/tests/behat/bootstrap/FeatureContext.php (hunk 1 above)
Comment (barbun) [inline build/tests/behat/bootstrap/FeatureContext.php]: If we are running under a driver other than Selenium then it just needs to skip but not fail.

**Pull request merged**